### PR TITLE
Integration of Refactored Classes

### PR DESCRIPTION
**Pull request by janagoe**
# Proposed Changes

Refactored classes include
- `AgvLogicalNs3` -> `AmrLogicalAgent`
- `TransportOrderApplicationNs3` -> `MaterialFlowLogicalAgent`
- `Kinematics` -> `AmrKinematics`

There are a lot of changes. However, this was inevitable due to many dependencies and, otherwise, occurring compilation errors. 

# Definition of Done (optional)
- [x] Feature is implemented
- [x] No known bugs that stops the correct execution
- [ ] CI / CD pipeline passed
- [x] The code guidelines were followed
- [ ] Unit and / or Integration tests for the new feature
- [ ] New feature was added to the documentation

### CODE REVIEW

File: daisi/src/path_planning/path_planning_manager.cpp [hunk 1]
```diff
@@ -135,25 +135,14 @@ uint64_t PathPlanningManager::getNumberOfNodes() {
...
-  properties.device_type = "AGV";
-  properties.model_name = "Mk42";
-  properties.model_number = 0;
-  properties.manufacturer = "FhG";
-
-  // Set dummy ability
-  properties.ability = cpps::amr::AmrStaticAbility(cpps::amr::LoadCarrier("package"), 100);
-
-  properties.kinematic = kinematics_;
+  cpps::AmrProperties properties("agv");
```
Comment (tseeland) [inline daisi/src/path_planning/path_planning_manager.cpp:138]: `amr`

File: daisi/src/path_planning/path_planning_manager.cpp [hunk 2]
```diff
@@ -590,9 +579,7 @@ void PathPlanningManager::parse() {
   auto min_velo = getParsed(float, "min_velo");
   auto max_acc = getParsed(float, "max_acc");
   auto min_acc = getParsed(float, "min_acc");
-  auto load_time_s = getParsed(float, "load_time_s");
-  auto unload_time_s = getParsed(float, "unload_time_s");
-  kinematics_ = cpps::Kinematics(max_velo, min_velo, max_acc, min_acc, load_time_s, unload_time_s);
+  kinematics_ = cpps::AmrKinematics(max_velo, min_velo, max_acc, min_acc);
```
Comment (tseeland) [inline daisi/src/path_planning/path_planning_manager.cpp:582]: use `AmrDescription` or an additional `AmrLoadHandlingUnit`

File: daisi/src/path_planning/path_planning_manager.h [hunk 3]
```diff
@@ -125,7 +123,7 @@ class PathPlanningManager final : public Manager<PathPlanningApplication> {
 
   consensus::ConsensusType consensus_type_ = consensus::ConsensusType::kNone;
   consensus::ConsensusSettings consensus_settings_;
-  cpps::Kinematics kinematics_;  //!< Kinematics of the simulated AGV
+  cpps::AmrKinematics kinematics_;  //!< Kinematics of the simulated AGV
```
Comment (tseeland) [inline daisi/src/path_planning/path_planning_manager.h:126]: see prevous comment

File: daisi/src/path_planning/agv_info.h [hunk 4]
```diff
@@ -30,7 +31,7 @@ namespace daisi::path_planning {
 struct AGVInfo {
   std::string id;
   std::string ip;
-  cpps::Kinematics kinematics;
+  cpps::AmrKinematics kinematics;
```
Comment (tseeland) [inline daisi/src/path_planning/agv_info.h:34]: use `AmrDescription` or additional `AmrLoadHandlingUnit` to store the same information.

File: daisi/src/path_planning/pickup_station.cpp [hunk 5]
```diff
@@ -94,8 +96,9 @@ void PickupStation::addToAGVOwnership(const message::NewAuthorityAGV &msg) {
   AGVInfo info;
   info.id = msg.agv_uuid;
   info.ip = msg.agv_ip;
-  info.kinematics = {msg.max_velocity,     msg.min_velocity, msg.max_acceleration,
-                     msg.min_acceleration, msg.load_time_s,  msg.unload_time_s};
+  info.kinematics = {msg.max_velocity, msg.min_velocity, msg.max_acceleration,
+                     msg.min_acceleration};
+  //, msg.load_time_s,  msg.unload_time_s};
```
Comment (tseeland) [inline daisi/src/path_planning/pickup_station.cpp:101]: AmrDescription/AmrLoadHandlingUnit

File: daisi/src/path_planning/pickup_station.cpp [hunk 6]
```diff
@@ -292,7 +294,7 @@ void PickupStation::initiateConsensus(const std::string &agv_id, RouteIdentifier
   }
   assert(!current_consensus_.has_value());
 
-  const cpps::Kinematics &kinematics = agv_ownership_.at(agv_id).kinematics;
+  const cpps::AmrKinematics &kinematics = agv_ownership_.at(agv_id).kinematics;
```
Comment (tseeland) [inline daisi/src/path_planning/pickup_station.cpp:297]: AmrDescription/AmrLoadHandlingUnit

File: daisi/src/path_planning/pickup_station.cpp [hunk 7]
```diff
@@ -307,7 +309,7 @@ void PickupStation::initiateConsensus(const std::string &agv_id, RouteIdentifier
   current_consensus_.emplace(ConsensusData{agv_id, intersection_times});
 
   consensus_->findConsensus(
-      intersection_times, kinematics.getLoadTime(),
+      intersection_times, 42,  // kinematics.getLoadTime(),
```
Comment (tseeland) [inline daisi/src/path_planning/pickup_station.cpp:312]: AmrDescription/AmrLoadHandlingUnit

**Review by tseeland — COMMENTED** [review]
All in all mostly naming suggestions.

In path planning `Kinematics` was replaced by `AmrKinematics` which does not contain durations for load handling. Those can be found in `AmrLoadHandlingUnit`. If path planning should work the same way as it did previously `AmrLoadHandlingUnit` must be used.

**Review by tseeland — COMMENTED** [review]
nice addition to cpps/verification.py

File: daisi/src/cpps/logical/algorithms/disposition/iterated_auction_disposition_participant.cpp [hunk 8]
```diff
@@ -121,7 +131,7 @@ bool IteratedAuctionDispositionParticipant::process(const WinnerNotification &wi
   return true;
 }
 
-void IteratedAuctionDispositionParticipant::calculateBids(const AuctionParticipantState &state) {
+void IteratedAuctionDispositionParticipant::calculateBids(AuctionParticipantState &state) {
```
[the remark this answers is not in the document]
Comment (janagoe) [reply daisi/src/cpps/logical/algorithms/disposition/iterated_auction_disposition_participant.cpp:134]: for readability

File: daisi/src/path_planning/path_planning_application.h [hunk 9]
```diff
@@ -47,12 +47,12 @@ struct PathPlanningApplication final : public ns3::Application {
   void initDeliveryStation(const DeliveryStationInfo &info,
                            const consensus::ConsensusSettings &settings);
   void initAGVLogical(const consensus::ConsensusSettings &settings, bool first_node);
-  void initAGVPhysical(const cpps::AgvDataModel &data_model, int id_friendly);
+  void initAGVPhysical(const cpps::AmrDescription &data_model, int id_friendly);
```
[the remark this answers is not in the document]
Comment (janagoe) [reply daisi/src/path_planning/path_planning_application.h:50]: the mobility model will now be created in the agv physical basic constructor

File: daisi/src/cpps/logical/amr/amr_logical_agent.cpp [hunk 10]
```diff
@@ -213,7 +256,42 @@ void AmrLogicalAgent::logAmrInfos() {
...
+  sola_->addService(service);
+}
+
+void AmrLogicalAgent::ExecutionState::setNextOrder() {
+  order_index++;
+  order_state = OrderStates::kCreated;
+
+  if (task.getOrders().size() > order_index) {
+    order_index = -1;
+  }
```
[the remark this answers is not in the document]
Comment (janagoe) [reply daisi/src/cpps/logical/amr/amr_logical_agent.cpp]: `-1` indicates that all orders from a task are finished and that the next task can be sent. 

A `kInvalid` should never get sent.

**Review by tseeland — APPROVED** [review]
Looks fine to me. Nice work.

**Pull request merged**